### PR TITLE
boringssl 0.20240913.0

### PR DESCRIPTION
**Pull request by davidben**

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @agl, @bob-beck, modules you maintain (boringssl) have been updated in this PR. Please review the changes.

**Comment by davidben** [comment]
> (15:45:31) ERROR: file:///workdir/modules/googletest/1.15.2/MODULE.bazel:68:20: name 'use_repo_rule' is not defined

Ah fun. I guess the new googletest is incompatible with Bazel 6? I guess we should probably just drop that from our CI config and assume that anyone using Bazel 6 is not using bzlmod and BCR. Will update this.

File: modules/boringssl/0.20240913.0/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  # Work around https://github.com/bazelbuild/bazel/issues/10472
+  verify_targets_on_macos:
+    name: Verify build targets on macOS
+    platform: macos
+    bazel: ${{ bazel }}
+    build_targets: *build_targets
+    build_flags:
+      - '--cxxopt=-std=c++14'
+      - '--sandbox_block_path=/usr/local'
+  run_test_module:
```
Comment (meteorcloudy) [inline modules/boringssl/0.20240913.0/presubmit.yml:27]: The test module is setup differently, you should put this task under `bcr_test_module`, see https://github.com/bazelbuild/bazel-central-registry/tree/main/docs#test-module

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

File: modules/boringssl/0.20240913.0/presubmit.yml (hunk 1 above)
Comment (davidben) [reply modules/boringssl/0.20240913.0/presubmit.yml:30]: This is just a copy of the presubmit that someone else wrote. Did the conventions change? Updated, I think.

File: modules/boringssl/0.20240913.0/presubmit.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    name: Verify build targets on macOS
+    platform: macos
+    bazel: ${{ bazel }}
+    build_targets: *build_targets
+    build_flags: &macos_workaround
+      - '--cxxopt=-std=c++14'
+      - '--sandbox_block_path=/usr/local'
+bcr_test_module:
+  module_path: util/bazel-example
+  matrix: *matrix
```
Comment (meteorcloudy) [inline modules/boringssl/0.20240913.0/presubmit.yml:29]: Missing `tasks:`

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

File: modules/boringssl/0.20240913.0/presubmit.yml (hunk 1 above)
Comment (davidben) [reply modules/boringssl/0.20240913.0/presubmit.yml:30]: PR #1833

**Review by meteorcloudy — APPROVED** [review]
Thanks!

File: modules/boringssl/0.20240913.0/presubmit.yml (hunk 1 above)
Comment (meteorcloudy) [reply modules/boringssl/0.20240913.0/presubmit.yml]: Ah, I see, this is missed from the review of https://github.com/bazelbuild/bazel-central-registry/pull/1833, /cc @mmorel-35 @fmeum

**Pull request merged**